### ISSUE TITLE
clamav-clamonacc has significant impact / low performance / additional option for FANOTIFY onaccess on write/execute only?

### ISSUE THREAD

**Issue by amsnek** [Cisco-Talos/clamav#1024]
Describe the bug
----------------

This is more a feature request. Clamonacc is impractical to use on more active systems, With on-access prevention it may even DoS a System quickly. It would be usefull having more control over what is actually scanned with onaccess. 

Following and experimenting a bit of the discussion on: https://linuxfun.org/en/2020/12/12/clamonacc-clamd-cpu100percent-en/
-> setting clamonacc (or having the option todo so) to only scan writes and execute file operation would remove a lot of the pressure and increase performance depending on use-case. With additional intervall based scans, one could ensure that existing files are ok, while new written files (or executed) are scanned with the on-access scanner. 

This could be acchieved with modifying clamonacc/fanotif/fanotif.c / (*ctx)->fan_mask |= FAN_ACCESS | FAN_OPEN

How to reproduce the problem
----------------------------

Use clamav-clamonacc on busy systems and/or with on-access prevention.

**Comment by rampaq** [comment]
I agree, it would be helpful. Even though it would not prevent e.g. opening malicious PDF files, the current OnAccess is unusable with OnAccessPrevention=yes - large amount of lag (minutes to open firefox) is introduced.
